conan/internal/runner/docker.py

```diff
@@ -0,0 +1,268 @@
+from collections import namedtuple
+import os
+import json
+import platform
+import shutil
+import yaml
+from conan.api.model import ListPattern
+from conan.api.output import Color, ConanOutput
+from conan.api.conan_api import ConfigAPI
+from conan.cli import make_abs_path
+from conan.internal.runner import RunnerException
+from conans.client.profile_loader import ProfileLoader
+from conans.errors import ConanException
+from conans.model.version import Version
+
+
+def config_parser(file_path):
+    Build = namedtuple('Build', ['dockerfile', 'build_context', 'build_args', 'cache_from'])
+    Run = namedtuple('Run', ['name', 'environment', 'user', 'privileged', 'cap_add', 'security_opt', 'volumes'])
+    Conf = namedtuple('Conf', ['image', 'build', 'run'])
+    if file_path:
+        def _instans_or_error(value, obj):
+            if value and (not isinstance(value, obj)):
+                raise Exception(f"{value} must be a {obj}")
```

Comment:
```suggestion
                raise ConanException(f"docker runner configfile syntax error: {value} must be a {obj.__name__}")
```

Else with a (super wrong) configfile like 
```yaml
image:
  - 1
  - 2
  - 3
```

I get ugly tracebacks like
```


ERROR: Traceback (most recent call last):
  File "/Users/ruben/coding/conan/conan/cli/cli.py", line 291, in main
    cli.run(args)
  File "/Users/ruben/coding/conan/conan/cli/cli.py", line 193, in run
    command.run(self._conan_api, args[0][1:])
  File "/Users/ruben/coding/conan/conan/cli/command.py", line 164, in run
    info = self._method(conan_api, parser, *args)
  File "/Users/ruben/coding/conan/conan/cli/commands/create.py", line 82, in create
    return runner_instance(conan_api, 'create', profile_host, profile_build, args, raw_args).run()
  File "/Users/ruben/coding/conan/conan/internal/runner/docker.py", line 102, in __init__
    self.configfile = config_parser(host_profile.runner.get('configfile'))
  File "/Users/ruben/coding/conan/conan/internal/runner/docker.py", line 29, in config_parser
    image=_instans_or_error(runnerfile.get('image'), str),
  File "/Users/ruben/coding/conan/conan/internal/runner/docker.py", line 24, in _instans_or_error
    raise Exception(f"{value} must be a {obj}")
Exception: [1, 2, 3] must be a <class 'str'>

ERROR: [1, 2, 3] must be a <class 'str'>
```

This also improves the erorr message a bit so that users are not super confused (even though getting what key is failing is harder)